subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -589,6 +589,395 @@ int ull_adv_aux_reset(void)
 	return 0;
 }
 
+uint8_t ull_adv_aux_hdr_set_clear(struct ll_adv_set *adv,
```

Comment:
This function is still quite large, and IMO should be split up even further

Reply:
Suggestions?

Reply:
I would think good candidates for a function would include (note I'm not 100% sure about naming convention we should use for these static inline functions):
lines 613-627: adv_acquire_lll_aux
lines 630-645: adv_fill_pri_pdu
lines 648-678: adv_fill_sec_pdu
lines 775-782: adv_calc_pri_pdu_len
lines 784-812: adv_calc_sec_pdu_len
lines 844-868: adv_fill__sec-pdu_sync
lines 872-892: adv_fill_aux_pri_pdu
lines 895-911: adv_fill_aux_sec_pdu

Reply:
perhaps this could be also called smth like `ull_adv_aux_configure` - the same code can be used to basically create an aux for advertising instance without adding or removing sync info. it would be used in e.g. `ll_adv_params_set` when configured instance needs aux (scannable or connectable) or later in `ll_adv_aux_ad_data_set` where aux is now created by pretty much the same code.

Reply:
Code blocks that was repeated across ull_adv, ull_adv_aux, and ull_adv_sync have been refactored to be reused. The function has not been split up further due to the fact that this function needs to be refactored based on how scan response and chaining is to be handled. 

Any further refactoring/split is viable only after(/ are part of) scannable mode support.